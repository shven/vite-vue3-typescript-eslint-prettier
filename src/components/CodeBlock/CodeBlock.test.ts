import { mount } from '@vue/test-utils';
import Component from './VueCodeBlock.vue';

describe('VueCodeBlock.vue', () => {
    const wrapper = mount(Component, {
        slots: {
            default: 'Slot content goes here'
        }
    });

    it('renders slot content', () => {
        expect(wrapper.text()).toContain('Slot content goes here');
    });

    it('Match snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});
