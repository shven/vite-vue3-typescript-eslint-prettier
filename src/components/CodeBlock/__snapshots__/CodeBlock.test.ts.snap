// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VueCodeBlock.vue > Match snapshot 1`] = `
<pre
  class="code-block"
  data-v-e6e8e1f5=""
>
          
  <code
    class="code-block__code"
    data-test="slot"
    data-v-e6e8e1f5=""
  >
    
    Slot content goes here
    
  </code>
  
        
  <button
    class="code-block__button"
    data-test="button"
    data-v-e6e8e1f5=""
  >
    
            
    <div
      class="icon icon--small"
      data-v-836caaa4=""
      data-v-e6e8e1f5=""
    >
      <!---->
    </div>
    
            Copy
        
  </button>
  
        
  <!--v-if-->
  
    
</pre>
`;
